File: twoSum.js
function twoSum(arr, target) {
    const counter = {}    
    for(let i = 0; i < arr.length; i++){
        const potentialMatch = target - arr[i]
        if(counter[potentialMatch]){
            return [arr.indexOf(potentialMatch), i]
        }
        counter[arr[i]] = true
    }
};

File: twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        catalog = {}
        for i in range(len(nums)):
            potential_match = target - nums[i]
            if potential_match in catalog:
                return [nums.index(potential_match), i]
            else:
                catalog[nums[i]] = True

